.gitlab/ci/.gitlab-ci.sdk-nightly.yml

```diff
@@ -235,26 +281,180 @@ demisto-sdk-nightly:run-end-to-end-tests:
+    CLOUD_SERVERS_FILE: "xsoar_ng_servers_path"
+    CLOUD_API_KEYS: $XSOAR_NG_API_KEYS
+    GCS_SOURCE_BUCKET: "${GCS_PRODUCTION_XSOAR_SAAS_BUCKET}"
+    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_XSOAR}"
+    ARTIFACTS_FOLDER_INSTANCE: "${ARTIFACTS_FOLDER_XSOAR}/instance_saas"
+    ENV_RESULTS_PATH: "${ARTIFACTS_FOLDER_XSOAR}/env_results.json"
+    GCS_MACHINES_BUCKET: "${TEST_XDR_PREFIX}marketplace-saas-dist-dev/upload-flow/builds-xsoar-ng"
+    MARKETPLACE_NAME: "xsoar_saas"
+    NON_REMOVABLE_PACKS: "Base"
+    NIGHTLY: "true"
+  needs: ["demisto-sdk-nightly:xsoar-saas-prepare-testing-bucket"]
+  stage: run-instances
+  script:
+    - EXIT_CODE=0
+
+    - section_start "Create Artifacts instance directory" --collapsed
+    - |
+      echo "Creating Artifacts instance directory: ${ARTIFACTS_FOLDER_INSTANCE}"
+      [ ! -d "${ARTIFACTS_FOLDER_INSTANCE}" ] && mkdir -p "${ARTIFACTS_FOLDER_INSTANCE}"
+      echo "${INSTANCE_ROLE}" > "${ARTIFACTS_FOLDER_INSTANCE}/instance_role.txt"
+    - section_end "Create Artifacts instance directory"
+
+    - !reference [.download-demisto-conf]
+    - !reference [.secrets-fetch]
+    - !reference [.lock-machine]
+
+    - !reference [.uninstall-packs-and-reset-bucket-cloud]
+    - !reference [.export_cloud_machine_constants]
+
+    - section_start "Copy env results to artifacts folder" --collapsed
+    - |
+      # workaround for the hard-coded value in the sdk
+      cp "${ARTIFACTS_FOLDER}/filter_file.txt" "./artifacts/filter_file.txt"
+    - section_end "Copy env results to artifacts folder"
+
+    - section_start "Create and Upload Entities to XSOAR Instance" --collapsed
+    - python3 Tests/sdknightly/create_entities_for_nightly_sdk.py HelloWorld --artifacts-folder $ARTIFACTS_FOLDER
+    - python3 Tests/sdknightly/create_entities_for_nightly_sdk.py AzureSentinel --artifacts-folder $ARTIFACTS_FOLDER
+    - demisto-sdk upload -i Packs/AzureSentinel --insecure -z
+    - demisto-sdk upload -i Packs/HelloWorld/TestPlaybooks/playbook-HelloWorld-Test.yml --insecure
+    - section_end "Create and Upload Entities to XSOAR Instance"
+
+    - section_start "Download Entities from XSOAR Instance" --collapsed
+    - demisto-sdk download -lf --insecure
+    - demisto-sdk download -o Packs/HelloWorld/ -i CommonServerUserPowerShell --insecure -f
+    - demisto-sdk download -o Packs/HelloWorld/ -i HelloWorld-Test --insecure -f
+    - section_end "Download Entities from XSOAR Instance"
+
+    - section_start "Install Packs" --collapsed
+    - cat "$ARTIFACTS_FOLDER/filter_file.txt" 
+    - ./Tests/scripts/install_content_and_test_integrations.sh || EXIT_CODE=$?
+    - cp -f "${ARTIFACTS_FOLDER}/conf.json" Tests/conf.json
+    - section_end "Install Packs"
+
+    - section_start "Wait Until Server Ready" --collapsed
+    - echo Going to sleep for 15 minutes to allow server finish indexing
+    - sleep-with-progress 900 30 "Sleeping... " 150
+    - echo "Done sleeping!"
+    - section_end "Wait Until Server Ready"
```

Comment:
its needed here?

Reply:
Same answer as above

Reply:
not needed for cloud machine IMO, but double check